pkg/registry/apis/query/query.go

```diff
@@ -135,6 +135,17 @@ func (b *QueryAPIBuilder) handleQuerySingleDatasource(ctx context.Context, req d
 			}
 		}
 	}
+
+	// Create a response object with the error when missing (happens for client errors like 404)
+	if rsp == nil && err != nil {
+		rsp = &backend.QueryDataResponse{Responses: make(backend.Responses)}
+		for _, q := range req.Request.Queries {
+			rsp.Responses[q.RefID] = backend.DataResponse{
+				Status: backend.Status(code),
+				Error:  err,
+			}
+		}
+	}
 	return rsp, err
```

Comment:
should the error also be set on the response in all cases?

Reply:
the error is already handled upstream -- the missing part was the status code getting lost and just returning an empty response (and error that did not point to any query)

Reply:
what about in situations like line 128? is it intentional that the validation error is overwriting `err` and returned on line 149?

Reply:
It is intentional -- though I am not sure it is a good idea yet.